hpx/util/itt_notify.hpp

```diff
@@ -219,19 +220,26 @@ namespace hpx { namespace util { namespace itt
     };
 
     //////////////////////////////////////////////////////////////////////////
+    struct itt_domain_tag {};
+
     struct domain
     {
-        HPX_EXPORT domain(char const* name);
+        HPX_NON_COPYABLE(domain);
+
+        HPX_EXPORT domain();
+        HPX_EXPORT ~domain();
 
-        ___itt_domain* domain_;
+        HPX_EXPORT static hpx::util::thread_specific_ptr<
+                ___itt_domain, itt_domain_tag
+            > domain_;
     };
```

Comment:
I'm not sure if making the `thread_specific_ptr` 'visible' to other modules through `HPX_EXPORT` is really necessary. Keeping it fully contained in the source file (itt_notify.cpp) without exposing it should be possible and sufficient.

Reply:
You're right, the `HPX_EXPORT` is not necessary. Removed.

Reply:
That's not what I had in mind (although that's part of it). Sorry if I was not sufficiently clear with my comments.

What I meant is to remove the `domain_` variable from the class entirely and make it `static` to the file it is defined in. There is no need for the user to see this at all.

Reply:
This comment completely slipped past me, sorry. Just wanted to say that I've seen it now and I'll take it into account once I look at this again.